### PR TITLE
Refactor initial db setup as the first step in a migration

### PR DESCRIPTION
**Pull request by SamHames**
- Migrations are a linear sequence of changes identified by an
  integer offset into a list of function for up/down.
- Migrations are to be applied in order.
- Add a database created with 0.10.0 as a test resource for basic
  migration logic.
- Move and refactor script locations to make the intent clearer in 
  future changes.

### CODE REVIEW

**Comment by rstuart85** [comment]
Are these failed tests expected?

File: caterpillar/storage/_sqlite_migrations.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+
+
+down_migrations = {
+
+}
+
+
+V0_10_0_schema = """
+/* pragma note: the page_size is set for the whole database, and can only change with
+and expensive vaccuum operation - it also must be run before anything else is done
```
Comment (rstuart85) [inline caterpillar/storage/_sqlite_migrations.py]: and = an

File: caterpillar/storage/_sqlite_migrations.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+with the database. So this line must come first! */
+pragma page_size = 4096; -- current recommended value for SQLite.
+
+/* Set write ahead log mode for all connections that will use this database. The WAL
+allow concurrent readers and writers, unlike the default journal mode that locks readers
```
Comment (rstuart85) [inline caterpillar/storage/_sqlite_migrations.py]: allow = allows

**Comment by SamHames** [comment]
I think the test failures are a packaging issue - looking into it now.

**Comment by SamHames** [comment]
It would be a significant investment of effort to create a Django style infrastructure. If we have migrations occuring at the same rate/style as a django project I'd be quite concerned. 

If we want to go down that route, it would be better to use something like [Alembic](http://alembic.zzzcomputing.com/en/latest/), or maybe one of the task runner libraries like [doit](http://pydoit.org/) to manage instead of reinventing the wheel.

As an aside, Django's migrations are dependencies trees, not linear -- the alphabetical sort is just the starting point for resolving the migration order.

**Comment by rstuart85** [comment]
Okay. I didn't mean for it to get complicated. I just meant for it to be more modular. Right now we will have a bunch of functions and random SQL code in 1 file that will be our migrations. Might want to go one step further, create a migration class, have them in different files, and join it all together in an `__init__` file.

**Comment by rstuart85** [comment]
For what it's worth, we will probably need to expose a migration command (or similar) in chrysalis. That's when something like doit *might* become useful. But even then, seems much easier to write our own command.

**Comment by SamHames** [comment]
This refactored a migration into a class, and rearranged for readability. I've held off on one file per migration for the moment - we can always do that later if needed.

I've also kept the migration order explicitly defined in a list - this will be useful later if we ever want/need to provide a fast path migration that merges a few expensive steps.

**Pull request merged**